src/Common/src/System/Security/Cryptography/DSACng.SignVerify.cs

```diff
@@ -24,7 +24,7 @@ public override byte[] CreateSignature(byte[] hash)
                 hash = AdjustHashSizeIfNecessary(hash);
                 unsafe
                 {
-                    byte[] signature = CngCommon.SignHash(_keyHandle, hash, AsymmetricPaddingMode.None, null, hash.Length * 2);
+                    byte[] signature = CngCommon.SignHash(GetDuplicatedKeyHandle(), hash, AsymmetricPaddingMode.None, null, hash.Length * 2);
```

Comment:
We should be disposing of the safe handles created by GetDuplicatedKeyHandle rather than leaving them for finalization.